pkg/ccl/importccl/import_stmt_test.go

```diff
@@ -2127,6 +2127,17 @@ func TestImportIntoCSV(t *testing.T) {
 		)
 	})
 
+	// Tests the case where we create table columns in specific order while trying
+	// to import data from csv where columns order is different and import expression
+	// defines in what order columns should be imported to align with table definition
+	t.Run("target-cols-reordered", func(t *testing.T) {
+		sqlDB.Exec(t, "CREATE TABLE t (a STRING NOT NULL, b INT NOT NULL)")
+		defer sqlDB.Exec(t, `DROP TABLE t`)
+
```

Comment:
i'm inclined to also include specifying a strict subset of the columns to ensure not only that we're ordering the set correctly but also using the right set. maybe something like 
```
 		sqlDB.Exec(t, "CREATE TABLE t (a INT PRIMARY KEY, b INT, c STRING)")
 		defer sqlDB.Exec(t, `DROP TABLE t`)
    data := `"c is a string", 1`

		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			if r.Method == "GET" {
				_, _ = w.Write([]byte(data))
			}
		}))
		defer srv.Close()
```
and then swap `(b, a)` for `(c, a)` and `testFiles.files[0]` for `srv.URL` below?

Reply:
done